Dev

ssoyeonnii opened the pull request:
랜딩페이지 추가 ~ 전체 UI 수정 및 회원정보수정 페이지 추가

ssoyeonnii committed:
feat : 인덱스 페이지 추가, 폰트 CDN 링크

ssoyeonnii committed:
feat : 홈페이지 랜딩페이지인 index페이지추가하여 route 컴포넌트 추가

ssoyeonnii committed:
feat : 회원탈퇴,로그아웃 버튼 모달창으로 이동

ssoyeonnii committed:
style : css수정

ssoyeonnii committed:
json파일 추가

ssoyeonnii committed:
style : css 수정

ssoyeonnii committed:
holidayutils 비공식 휴일 리스트 추가

ssoyeonnii committed:
holidayutils import삭제 및 근무,휴무일 분기 추가

ssoyeonnii committed:
회원정보수정페이지 추가

ssoyeonnii committed:
로그인,회원가입 폼 수정 및 회원가입 시 로그인 처리

ssoyeonnii committed:
회원정보관련 모달 추가

ssoyeonnii committed:
근무 셀 ui 수정

ssoyeonnii committed:
근무설정 모달 ui 및 기능 수정
- 회원정보수정, 로그아웃 버튼 삭제

ssoyeonnii committed:
홈페이지 랜딩페이지

ssoyeonnii committed:
routes추가
-UserUpdate 회원정보수정페이지

ssoyeonnii committed:
랜딩페이지 스크롤바 색상 적용
- 브라우저 테마 별

ssoyeonnii committed:
달력 헤더 수정
-근무설정,회원정보 모달 버튼

ssoyeonnii committed:
회원탈퇴 페이지 style > class부여

ssoyeonnii committed:
- defaultconfig 근무,휴무일 기본 값 부여
- 오늘날짜의 월로 이동하는 함수 추가
- 사용자 정보 모달 컴포넌트 추가

ssoyeonnii committed:
태그 별 css 수정

ssoyeonnii committed:
로그아웃, 탈퇴 함수 사용안해서 코드 삭제

ssoyeonnii committed:
달력 선택 모달창 생성
-기존 datepicker에서 변경

ssoyeonnii committed:
달력 선택 모달창의 css class추가

ssoyeonnii committed:
달력 날짜 선택 모달창 생성하여 관련 로직 변경
-기존datepicker에서 변경

ssoyeonnii committed:
-랜딩페이지 모바일 버전 css틀어짐 수정

ssoyeonnii committed:
회원가입 시 이름 패턴 alert창 오타

ssoyeonnii committed:
근무설정 모달 창 버튼 padding class 수정

ssoyeonnii committed:
랜딩 페이지 UI 변경
-기존 캘린더의 문제점 UI
-dayf 해결책 제시 UI

ssoyeonnii committed:
캘린더 버튼 Ui변경
-달력 월 이동 버튼
-오늘 버튼 ,오늘 날짜
-달력 월 -> 연월 출력형식 변경

ssoyeonnii committed:
-버튼 ui 변경 및 페이지 제목과 컨텐츠 간격 조정

ssoyeonnii committed:
로그인 버튼 영역 margin bottom class명 오타

ssoyeonnii committed:
오늘 날짜 표시 css 변경

ssoyeonnii committed:
사용자 정보 모달 삭제
-툴팁형태로 출력

ssoyeonnii committed:
로그인 페이지 margin 수정

ssoyeonnii committed:
회원탈퇴 페이지 버튼 css수정

ssoyeonnii committed:
회원탈퇴 버튼 css수정

ssoyeonnii committed:
react-oauth/google 설치

ssoyeonnii committed:
구글 이메일 로그인 연동
-최초 로그인 시 db저장
-구글 로그인 시 Session(googleuser)저장
-구글유저, 일반 유저 탈퇴 방식 변경

ssoyeonnii committed:
프론트 auth-code 플로우로 전환

ssoyeonnii committed:
백엔드 서버 node.js 구축 완료
-refresh 토큰 저장

ssoyeonnii committed:
#8
-구글 로그인 시 scope calendar.events 추가
-테스트 이벤트 추가 버튼 생성 및 테스트
-server.js : .env파일 찾지 못해 절대 경로로 설정

mesoftkor commented:
[백엔드 서버 node.js 구축 완료](https://github.com/ssoyeonnii/dayf/pull/6/commits/d2d070c5a52ce49b03e9864c8f17ee69a378e2bd)
갑자기 node.js 백엔드 서버는 뭘까요?

ssoyeonnii committed:
#10
1.톱니바퀴(근무설정) 아이콘 툴팁 메뉴 작업완료
1-1. Google Calendar 미연동 상태
-근무 설정 버튼
-Google Calendar 연동 버튼
1-2. Google Calendar 연동 완료 상태
-근무 설정 버튼
-연동된 Google 계정 이메일 표시
-Google Calendar 일정등록 버튼
-Google 연동 계정 변경 버튼
-Google 연동 해제 버튼

ssoyeonnii committed:
#10
GoogleLoginButton.jsx의 session storage googleuser 저장 삭제

ssoyeonnii committed:
#10
2.Google Calendar 연동 버튼
2-1. dayf 회원가입 사용자
버튼 클릭 시 Google OAuth 인증 팝업
2-2. Google 소셜 로그인 사용자
로그인 시 Google OAuth 팝업창에서 Calendar 접근 허용 체크 하지않을 경우 해당 버튼을 통해 연동 진행
2-3. Google Calendar 연동 성공 시
톱니바퀴(근무설정) 아이콘 툴팁에 연동된 Google 계정 이메일 표시
연동 완료 안내 메세지 출력
2-4. 연동 실패 시
OAuth 인증 실패 에러 메시지 표시
error log 저장

ssoyeonnii committed:
#10
-Dayf 계정 ID 로그인 후 google 연동 체크 시 session의 google_email도 체크
 -이미 Dayf 계정 ID로 등록된 google 계정으로 캘린더 연동 시 에러 메세지 변수에 저장하여 로그 저장
함수 호출

ssoyeonnii committed:
#10
-로그인 시 연동된 gmail 값 체크
-회원가입 시 이미 가입된 google 계정인지 체크
-회원탈퇴 시 google 관련 session remove

ssoyeonnii committed:
#10
-테스트 이벤트 추가(구글캘린더) 버튼 삭제

ssoyeonnii committed:
#10
-google 유저일 경우 회원정보수정 버튼 나타나지 않음
-db연결하는 함수 삭제
-session의 구글 유저 체크 로직 중복으로 삭제

ssoyeonnii committed:
#10
-근무설정 값 입력 폼 유효성검사 세분화 로직 추가

ssoyeonnii committed:
#10
-구글 소셜로그인 버튼 수정
-session추가

ssoyeonnii committed:
#10
-UserInfoModal.jsx 삭제
-googleuser session 저장 분기 삭제
